### PR TITLE
Refactor MNS private API

### PR DESCRIPTION
**Pull request by davidmanzanares**
Regarding https://github.com/CartoDB/carto-vl/issues/407

### CODE REVIEW

File: src/client/windshaft.js [hunk 1]
```diff
@@ -322,9 +329,8 @@ export default class Windshaft {
         if (!response.ok) {
             throw new Error(`Maps API error: ${JSON.stringify(layergroup)}`);
         }
-        this._URLTemplates = layergroup.metadata.tilejson.vector.tiles;
         return {
-            url: getLayerUrl(layergroup, LAYER_INDEX, conf),
+            urlTemplates: layergroup.metadata.tilejson.vector.tiles,
```
Comment (jgoizueta) [inline src/client/windshaft.js]: This seems more sensible now

File: src/renderer/schema.js [hunk 2]
```diff
@@ -1,40 +1,66 @@
...
-};
+
+export const IDENTITY = {};
 
 /*
-const schema = {
-    columns: ['temp', 'cat']
-}; */
+const mns = {
+  price:  new Set(['avg', 'max]),
```
Comment (jgoizueta) [inline src/renderer/schema.js]: They're not Sets anymore, right?

File: src/client/windshaft.js (hunk 1 above)
Comment (davidmanzanares) [reply src/client/windshaft.js]: yep, updated

File: src/renderer/schema.js (hunk 2 above)
Comment (davidmanzanares) [reply src/renderer/schema.js]: yep, updated

**Comment by davidmanzanares** [comment]
Tests finally pass, but performance may have regressed a lot when dealing with many properties, I'll review that case

**Comment by davidmanzanares** [comment]
I was worried about the performance impact of line https://github.com/CartoDB/carto-vl/pull/893/files#diff-70dfd5511c05de98bb8c59e31215cf55R89 but after profiling and taking a closer look, it seems that shouldn't cause trouble since it will only be called once.

**Comment by davidmanzanares** [comment]
Btw, I thought that this wouldn't solve the corner case of using a property only with global aggregations. I thought that it would still download the per-feature/tile data instead of just the metadata.

However, unless I'm missing something it works:

Instantiation request:
```
{
    "buffersize": {
        "mvt": 1
    },
    "layers": [
        {
            "type": "mapnik",
            "options": {
                "sql": "SELECT \"category\",\"amount\",\"cartodb_id\",\"the_geom_webmercator\" FROM spend_data ",
                "aggregation": {
                    "columns": {
                        "_cdb_agg_mode_category": {
                            "aggregate_function": "mode",
                            "aggregated_column": "category"
                        }
                    },
                    "dimensions": {},
                    "placement": "centroid",
                    "resolution": 1,
                    "threshold": 1,
                    "filters": {}
                },
                "dates_as_numbers": true,
                "metadata": {
                    "geometryType": true,
                    "columnStats": {
                        "topCategories": 32768,
                        "includeNulls": true
                    },
                    "sample": {
                        "num_rows": 1000,
                        "include_columns": [
                            "category",
                            "amount",
                            "cartodb_id"
                        ]
                    }
                }
            }
        }
    ]
}
```

Instantiation response:
```
...
},
                            "amount": {
                                "type": "number",
                                "min": 0.01,
                                "max": 15000,
                                "avg": 57.2069958615219,
                                "sum": 7188059.03000023
                            },
                            "cartodb_id": {
                                "type": "number",
                                "min": 22,
                                "max": 1178596,
                                "avg": 591151.2784241942,
                                "sum": 74278158134
                            },
                            "the_geom_webmercator": {
                                "type": "geometry"
                            }
...
```

MVT tile (converted to GeoJSON):
```
{
  "type": "FeatureCollection",
  "features": [
    {
      "type": "Feature",
      "geometry": {
        "type": "Point",
        "coordinates": [
          2.197265625,
          41.37680856570236
        ]
      },
      "properties": {
        "_cdb_agg_mode_category": "Moda y calzado",
        "_cdb_feature_count": 125650,
        "cartodb_id": 22
      },
      "id": 1
    }
  ]
}
```

Am I missing something? I thought that Maps API was unprepared for that.

cc: @jgoizueta

File: src/client/windshaft.js [hunk 3]
```diff
@@ -244,26 +244,28 @@ export default class Windshaft {
...
                 }
             });
 
         return aggregation;
     }
 
     _buildSelectClause (MNS) {
-        const columns = MNS.columns.map(name => schema.column.getBase(name))
-            .concat(['the_geom_webmercator', 'cartodb_id']);
+        const columns = Object.keys(MNS).concat(['the_geom_webmercator', 'cartodb_id']);
```
Comment (jgoizueta) [inline src/client/windshaft.js]: Is the following uniqueness filter needed now?

Comment (davidmanzanares) [reply src/client/windshaft.js]: I think so, the `cartodb_id` could be referenced within the viz and it would be appended now too

**Pull request merged**